sslh-select.c

```diff
@@ -319,6 +408,23 @@ void main_loop(int listen_sockets[], int num_addr_listen)
                             check_access_rights(in_socket, cnx[i].proto->service)) {
                             tidy_connection(&cnx[i], &fds_r, &fds_w);
                             res = -1;
+                        } else if (cnx[i].proto->fork) {
+                            if (!fork()) {
+                                struct connection *pcnx_i = &cnx[i];
+                                struct connection cnx_i = *pcnx_i;
+                                for (i = 0; i < num_addr_listen; i++)
+                                    close(listen_sockets[i]);
+                                for (i = 0; i < num_cnx; i++)
+                                    if (&cnx[i] != pcnx_i)
+                                        for (j = 0; j < 2; j++)
+                                            if (cnx[i].q[j].fd != -1)
+                                                close(cnx[i].q[j].fd);
+                                free(cnx);
+                                connect_proxy(&cnx_i);
+                                exit(0);
+                            }
```

Comment:
you need to process fork() == -1 to log an error message (you can't just drop connections without telling the user...)

Reply:
Good point!

`sslh-fork` doesn’t do it either though :-) That should probably be fixed. In fact, if it fails to fork a listener process, it will later try to invoke `kill(-1, ...)`, which might be a bit disastrous.